src/control/system/raft.go

```diff
@@ -164,15 +172,14 @@ func (d *dbData) applyMemberUpdate(op raftOp, data []byte) {
 	case raftOpUpdateMember:
 		cur, found := d.Members.Uuids[m.Member.UUID]
 		if !found {
-			d.log.Errorf("member update for unknown member %+v", m)
+			panic(errors.Errorf("member update for unknown member %+v", m))
```

Comment:
Some of these failure states seem pretty innocuous -- not sure I'd want to take down the entire server for it.

Reply:
I'm not 100% certain this is correct either, but after reading the issue I linked to above, it seemed like the best practice for dealing with an uncertain state. The result of applying a series of log entries should be a completely deterministic state, so if we get into a place where things seem inconsistent, it's probably a sign that we don't want this database to be participating as a replica.

Reply:
Upon further reflection, I realize that this is the wrong approach. I agree that we shouldn't take down the entire control plane process; that's a dumb idea. I just need to figure out how to forcefully remove the node from participation as a replica.

Reply:
I've flipped (or flopped?) back to keeping the panic()s in place for now. It's really not clear to me right now what it means to have a running control plane service that had a catastrophic failure while applying logs. The guidance from the authors of the raft implementation is that the FSM operations should all be in-memory (i.e. no exposure to i/o errors, etc), so a failure is most likely going to be due to programmer error or "can't happen" situations like -ENOMEM. I have to revisit this stuff in order to add multi-replica support, so I'll give more attention to complex failure scenarios then.

Reply:
That works for me, as long as it's on your radar. And I have no problem running with these panics here in the short-term, which will quickly expose bugs in this logic.